src/luascript.cpp

```diff
@@ -11994,6 +11990,141 @@ int LuaScriptInterface::luaPartySetSharedExperience(lua_State* L)
 	return 1;
 }
 
+// Spells
+int LuaScriptInterface::luaSpellCreate(lua_State* L)
+{
+	// Spell(words, name or id)
+	InstantSpell* spell;
+	if (isNumber(L, 2)) {
+		spell = g_spells->getInstantSpellById(getNumber<uint32_t>(L, 2));
+	} else if (isString(L, 2)) {
+		const std::string& stringArgument = getString(L, 2);
+		spell = g_spells->getInstantSpellByName(stringArgument);
+		if (spell == nullptr) {
+			spell = g_spells->getInstantSpell(stringArgument);
+		}
+	} else {
+		spell = nullptr;
+	}
+
+	if (spell) {
+		pushUserdata<InstantSpell>(L, spell);
+		setMetatable(L, -1, "Spell");
+	} else {
+		lua_pushnil(L);
+	}
+	return 1;
+}
+
+int LuaScriptInterface::luaSpellGetName(lua_State* L)
+{
+	// spell:getName()
+	InstantSpell* spell = getUserdata<InstantSpell>(L, 1);
+	if (spell) {
```

Comment:
Move the variable to inside the if (`if (InstantSpell* spell = getUserdata<InstantSpell>(L, 1))`) for better scoping.